### PR TITLE
Test: onthefly functionality

### PR DESCRIPTION
**Pull request by akhil-jha**
```
pytest tests/foreman/ui/test_ldap_authentication.py -k test_onthefly_functionality
============================================================================ test session starts ============================================================================
platform linux -- Python 3.6.8, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
plugins: services-1.3.1, mock-1.10.4, cov-2.10.1, forked-1.3.0, xdist-2.1.0
collecting ... 2020-09-05 13:09:37 - conftest - DEBUG - Collected 30 test cases
collected 30 items / 29 deselected / 1 selected                                                                                                                             

tests/foreman/ui/test_ldap_authentication.py .                                                                                                                        [100%]

================================================================= 1 passed, 29 deselected in 79.99 seconds ==================================================================
```

### CODE REVIEW

File: tests/foreman/ui/test_ldap_authentication.py [hunk 1]
```diff
@@ -1324,3 +1324,40 @@ def test_deleted_idm_user_should_not_be_able_to_login(auth_source_ipa, ldap_tear
         with raises(NavigationTriesExceeded) as error:
             ldapsession.user.search('')
         assert error.typename == "NavigationTriesExceeded"
+
+
+@tier2
+def test_onthefly_functionality(session, ipa_data, ldap_tear_down):
+    """IDM user will not be created automatically in Satellite if onthefly is
+    disabled
```
Comment (mirekdlugosz) [inline tests/foreman/ui/test_ldap_authentication.py:1332]: In test, you check that user can't log in. Would it make sense to explicitly check if user is not created?

**Review by mirekdlugosz — APPROVED** [review]
Code looks OK, left one question.
ACK

File: tests/foreman/ui/test_ldap_authentication.py (hunk 1 above)
Comment (akhil-jha) [reply tests/foreman/ui/test_ldap_authentication.py:1332]: The user will only be created in satellite after the user has been authenticated from LDAP(here IDM). So being said that, the user creation depends on the successful login.
Even if we `'account.onthefly_register': True`, until the user logs in, the account will not be created.

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Pull request merged**